crates/core/src/transport/connection_handler.rs

```diff
@@ -318,12 +322,34 @@ impl<S: Socket> UdpPacketsListener<S> {
                                         self.remote_connections.insert(remote_addr, remote_conn);
                                         continue;
                                     }
-                                    Err(mpsc::error::TrySendError::Full(packet)) => {
-                                        // Channel full, reinsert connection and restore packet_data
+                                    Err(mpsc::error::TrySendError::Full(_)) => {
```

Comment:
all the last changes in connection_handler and how we are handling channels are incorrect. Asynchronous mpsc channels have built back pressure, and messages are never dropped, there is a whole lot of complexity added here which is not proper usage of channels and is error prone.

Just use unboudned channels if this is an issue (and eventually find out why is an issue!).